experimental/rule_inference/README.md

```diff
@@ -0,0 +1,38 @@
+# PiranhaAgent 
+
+This code uses OpenAI's GPT-4 model to generate piranha rules in TOML format.
+
+## Requirements
+
+- Python 3.7+
+- OpenAI Python library
+- Tree-sitter Python library
+- Tree-sitter-languages Python library
+
+## Usage
+
+1. Set the `OPENAI_API_KEY` environment variable with your OpenAI API key.
+2. Run the script with the following command:
+
+```
+python main.py -s <source_file> -t <target_file> -l <language>
+```
+
+- `<source_file>`: Path to the original source code file.
```

Comment:
Also, could we have a mode where the user defines a toml file  
with entries - language, before file content, after file content 

(as a follow up Pr ? )

Reply:
yes